kroxylicious-kubernetes-api/src/main/resources/META-INF/fabric8/virtualkafkaclusters.kroxylicious.io-v1.yml

```diff
@@ -55,6 +55,9 @@ spec:
       schema:
         openAPIV3Schema:
           type: object
+          x-kubernetes-validations:
+            - rule: "self.spec.ingresses.all(ingress, (ingress.ingressRef.name.size() + self.metadata.name.size()) <= 51)"
```

Comment:
If we merge this, maybe we should also enforce that kpi.metadata.name and vkc.metadata.name are individually <= 49, it doesn't make much sense to let through names that cannot possibly be combined to create a valid ingress name

Reply:
Why do we assume they are concatenated into a single `hostname` (for want of another term the left most section of a dns name) can the ingress not be a subdomain of the vkc?

Reply:
not for the ingress clusterIP services, all you get is the service name which has to be a single DNS label, I don't think we can do anything with subdomains

for loadbalancer the DNS is going to be in the user's control so this restriction doesn't apply